File: test/ch21-benchmarks/bench-006-file-processing.py
#!/usr/bin/env python3
# BENCH-006: File Line Processing (Large File grep-like operation)
# Task: Read 100MB log file, count lines containing "error" (case-insensitive)
# Measures: Buffered I/O efficiency, string search performance

def count_error_lines(filename):
    """Count lines containing 'error' (case-insensitive)"""
    count = 0
    with open(filename, 'r') as f:
        for line in f:
            if 'error' in line.lower():
                count += 1
    return count

# Execute benchmark
result = count_error_lines('testdata/bench-006-logs-100mb.txt')
print(result)
# Expected: 126076

File: test/ch21-benchmarks/bench-006-file-processing.rs
// BENCH-006: File Line Processing - Rust
use std::fs::File;
use std::io::{BufRead, BufReader};

fn count_error_lines(filename: &str) -> std::io::Result<usize> {
    let file = File::open(filename)?;
    let reader = BufReader::new(file);
    
    let mut count = 0;
    for line in reader.lines() {
        let line = line?;
        if line.to_lowercase().contains("error") {
            count += 1;
        }
    }
    
    Ok(count)
}

fn main() -> std::io::Result<()> {
    let result = count_error_lines("testdata/bench-006-logs-100mb.txt")?;
    println!("{}", result);
    // Expected: 126076
    Ok(())
}
